locksmith/src/payment/paymentProcessor.ts

```diff
@@ -51,27 +51,26 @@ export class PaymentProcessor {
       const user = await this.findUserByPublicKey(publicKey)
       const stripeCustomerId = await getStripeCustomerIdForAddress(publicKey)
 
-      if (user && stripeCustomerId) {
+      // If we already have a stripe customer id
+      if (stripeCustomerId) {
         await this.stripe.customers.createSource(stripeCustomerId, {
           source: token,
         })
 
         return true
       }
-      if (user && !stripeCustomerId) {
-        const customer = await this.createStripeCustomer(
-          user.emailAddress,
-          token
-        )
-        return !!(await saveStripeCustomerIdForAddress(publicKey, customer.id))
-      }
-      return false
+
+      const customer = await this.createStripeCustomer(
+        token,
+        user ? user.emailAddress : ''
+      )
+      return !!(await saveStripeCustomerIdForAddress(publicKey, customer.id))
     } catch (e) {
       return false
     }
   }
 
-  createStripeCustomer(emailAddress: string, token: string) {
+  createStripeCustomer(token: string, emailAddress?: string) {
```

Comment:
I had to switch these 2 params becuase `emailAddress` needs to be optional.

Reply:
If this is the reason for taking this approach, lets take a different approach to achieving this. Lets create a type for the arguments so that the ordering doesnt matter.

Inside the body of the method you can proceed as you see fit